components/booking/Slots.tsx

```diff
@@ -0,0 +1,96 @@
+import { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import getSlots from "../../lib/slots";
+import dayjs, { Dayjs } from "dayjs";
+import isBetween from "dayjs/plugin/isBetween";
+import utc from "dayjs/plugin/utc";
+dayjs.extend(isBetween);
+dayjs.extend(utc);
+
+type Props = {
+  eventLength: number;
+  minimumBookingNotice?: number;
+  date: Dayjs;
+};
+
+const Slots = ({ eventLength, minimumBookingNotice, date, workingHours, organizerUtcOffset }: Props) => {
+  minimumBookingNotice = minimumBookingNotice || 0;
+
+  const router = useRouter();
+  const { user } = router.query;
+  const [slots, setSlots] = useState([]);
+  const [isFullyBooked, setIsFullyBooked] = useState(false);
+  const [hasErrors, setHasErrors] = useState(false);
+
+  useEffect(() => {
+    setSlots([]);
+    setIsFullyBooked(false);
+    setHasErrors(false);
+    fetch(
+      `/api/availability/${user}?dateFrom=${date.startOf("day").utc().startOf("day").format()}&dateTo=${date
+        .endOf("day")
+        .utc()
+        .endOf("day")
+        .format()}`
+    )
+      .then((res) => res.json())
+      .then(handleAvailableSlots)
+      .catch((e) => {
+        console.error(e);
+        setHasErrors(true);
+      });
+  }, [date]);
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has missing dependencies: 'handleAvailableSlots' and 'user'. Either include them or remove the dependency array.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689306)
```suggestion
  }, [date, handleAvailableSlots, user]);
```